drivers/project_scorecard/app/controllers/project_scorecard/warehouse_reports/scorecards_controller.rb

```diff
@@ -0,0 +1,162 @@
+###
+# Copyright 2016 - 2020 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+###
+
+module ProjectScorecard::WarehouseReports
+  class ScorecardsController < ApplicationController
+    include WarehouseReportAuthorization
+    include ArelHelper
+    before_action :set_projects, :set_project_groups, :set_current_reports
+    before_action :set_report, only: [:show, :edit, :update]
+
+    def index
+      start_date = Date.current.prev_month.beginning_of_month
+      end_date = Date.current.prev_month.end_of_month
+
+      @range = ::Filters::DateRange.new(start: start_date, end: end_date)
+    end
+
+    def for_project
+      project = project_scope.find(params[:project_id].to_i)
+      @name = project.name
+      @reports = reports_scope.where(project_id: project.id)
+      render :report_list
+    end
+
+    def for_group
+      project_group = project_group_scope.find(params[:project_group_id].to_i)
+      @name = project_group.name
+      @reports = reports_scope.where(project_group_id: project_group.id)
+      render :report_list
+    end
+
+    def create
+      @generate = create_parameters[:generate] == '1'
+      @email = create_parameters[:email] == '1'
+      @range = ::Filters::DateRange.new(date_parameters)
+      @range.validate
+      errors = @range.errors.messages.map { |k, v| "#{k}: #{v.join(', ')}".humanize }
+
+      @project_ids = params[:project]&.keys&.map(&:to_i) || []
+      @project_ids = project_scope.where(id: @project_ids).pluck(:id) # viewability allowlist
+
+      @project_group_ids = params[:project_group]&.keys&.map(&:to_i) || []
+      @project_group_ids = project_group_scope.where(id: @project_group_ids).pluck(:id) # viewability allowlist
```

Comment:
I think all of this can be delegated into `Filters::FilterBase`

Reply:
After some exploration, I decided not to handle the project ids via `Filters::FilterBase` -- they come back via a Hash (the form is the one from the DQ report, and  is generated via JS), and adding support for this corner case seemed inappropriate.

I also tore out the project group stuff since it isn't actually appropriate here.